test/NuGet.Clients.Tests/NuGet.CommandLine.Test/MSBuildUtilityTest.cs

```diff
@@ -231,6 +232,51 @@ public void TestGetMsbuildDirectoryFromPATHENV()
             }
         }
 
+        [Theory]
+        [InlineData('"')]
+        [InlineData('<')]
+        [InlineData('>')]
+        [InlineData('|')]
+        public void GetMsbuildDirectoryFromPath_PATHENVWithInvalidChars_Succeeds(char invalidChar)
+        {
+            using (var vsPath = TestDirectory.Create())
+            {
+
+                if (RuntimeEnvironmentHelper.IsMono)
+                { // Mono does not have SxS installations so it's not relevant to get msbuild from the path.
+                    return;
+                }
+
+                var msBuild159BinPath = Directory.CreateDirectory(Path.Combine(vsPath, "MSBuild", "15.9", "Bin")).FullName;
```

Comment:
1. `Directory.CreateDirectory` has a side effect, so I don't like that it's used with `.FullName` in the same expression. I'd like them to be separate so it's more clear to the reader what's going on.

2. Is it necessary to make the test directory look like a real MSBuild directory? When I read this line, my first reaction was we're now targeting d16 and we shouldn't rely on d15 being installed. After reading more carefully I see it doesn't need a d15.9 install, so a test which doesn't mislead readers would save time.